hoomd/md/external/field.py

```diff
@@ -0,0 +1,129 @@
+# Copyright (c) 2009-2021 The Regents of the University of Michigan
+# This file is part of the HOOMD-blue project, released under the BSD 3-Clause
+# License.
+
+# Maintainer: joaander / All Developers are free to add commands for new
+# features
+
+"""External field potentials."""
+
+import hoomd
+from hoomd.md import _md
+from hoomd.md import force
+from hoomd.data.parameterdicts import TypeParameterDict
+from hoomd.data.typeparam import TypeParameter
+
+
+class Field(force.Force):
+    """Common external field potential documentation.
+
+    Users should not invoke `Field` directly. Documentation common to all
+    external field potentials is located here. External potentials represent
+    forces which are applied to all particle in the simulation by an external
+    agent.
+    """
+
+    def _attach(self):
+        if isinstance(self._simulation.device, hoomd.device.CPU):
+            cls = getattr(_md, self._cpp_class_name)
+        else:
+            cls = getattr(_md, self._cpp_class_name + "GPU")
+
+        self._cpp_obj = cls(self._simulation.state._cpp_sys_def)
+        super()._attach()
+
+
+class Periodic(Field):
+    """One-dimension periodic potential.
+
+    `Periodic` specifies that an external force should be added to every
+    particle in the simulation to induce a periodic modulation in the particle
+    concentration. The modulation is one-dimensional and extends along the
+    lattice vector :math:`\\mathbf{a}_i` of the simulation cell. The force
+    parameters can be set on a per particle type basis. This potential can, for
+    example, be used to induce an ordered phase in a block-copolymer melt.
+
+    The external potential :math:`V(\\vec{r})` is implemented using the
+    following formula:
+
+    .. math::
+
+       V(\\vec{r}) = A * \\tanh\\left[\\frac{1}{2 \\pi p w} \\cos\\left(
+       p \\vec{b}_i\\cdot\\vec{r}\\right)\\right]
+
+    The coefficients above must be set per unique particle type.
+
+    .. py:attribute:: params
+
+        The `Periodic` external potential parameters. The dictionary has the
+        following keys:
+
+        * ``A`` (`float`, **required**) - Ordering parameter :math:`A` \
+            :math:`[\\mathrm{energy}]`.
+        * ``i`` (`int`, **required**) - :math:`\\vec{b}_i`, :math:`i=0, 1, 2`,
+            is the simulation box's reciprocal lattice vector in the :math:`i`
+            direction :math:`[\\mathrm{dimensionless}]`.
+        * ``w`` (`float`, **required**) - The interface width :math:`w`
+            relative to the distance :math:`2\\pi/|\\mathbf{b_i}|` between
+            planes in the :math:`i`-direction :math:`[\\mathrm{dimensionless}]`.
+        * ``p`` (`int`, **required**) - The periodicity :math:`p` of the
+            modulation :math:`[\\mathrm{dimensionless}]`.
```

Comment:
Something in this section isn't rendering right. Perhaps the backslash escape in the first line?
https://hoomd-blue--1013.org.readthedocs.build/en/1013/module-md-external-field.html#hoomd.md.external.field.Periodic
![image](https://user-images.githubusercontent.com/3943761/122591259-551a7a80-d028-11eb-8fe2-1c948739b6c8.png)